plugins/woocommerce/src/Blocks/TemplateOptions.php

```diff
@@ -0,0 +1,39 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Options;
+
+/**
+ * TemplateOptions class.
+ *
+ * @internal
+ */
+class TemplateOptions {
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates() {
+		$option_name           = Options::WC_BLOCK_USE_BLOCKIFIED_PRODUCT_GRID_BLOCK_AS_TEMPLATE;
+		$should_use_blockified = wc_bool_to_string( wc_current_theme_is_fse_theme() );
```

Comment:
Below we are using `! $should_use_blockified` in the conditional so if we use `wc_bool_to_string()` on this value it will never be falsey since `"no"` and `"yes"` will resolve to `true`. When we update the option value anyway we use `wc_bool_to_string()` before storing it anyway.

```suggestion
		$should_use_blockified = wc_current_theme_is_fse_theme();
```